docs/autoprocess.py

```diff
@@ -0,0 +1,397 @@
+            return querry_for_slug
+
+        elif content in categories:
+            if content == 'uncategorized':
+                filtered_dict = dict(
+                    pair for pair in processes_dict.items() if 'category' not in pair[1])
+                return filtered_dict
+
+            filtered_dict = dict(
+                pair for pair in processes_dict.items()
+                if 'category' in pair[1] and pair[1]['category'] == content)
+            return filtered_dict
+
+        raise ValueError("Invalid argument given to autoprocess directive.")
+
+    def get_process_definition_start(self, fname, slug):
+        """
+        Find the starting line of process definition - where slug is defined
+
+        :param path/str fname: Filename with processes
+        :param string slug: slug of the process
+        :return: line where the process definiton starts
+        :rtype: int
+        """
+        i = 1
+        with open(fname) as file_:
+            for line in file_:
+                if 'slug: ' + slug in line:
+                    return i
+                i += 1
+        # In case starting line is not found just return first line
+        return 1
+
+    def get_all_processes(self, process_dir, source_base_url):
+        """
+        Return a dictionary of all processes in `process_dir`
+
+        Dictionary key is an URL's to process source code and value is
+        process as it is loaded from yaml file.
+
+        :param path/str process_dir: Parent directory of all process files
+        :param url/string source_base_url: Base of the url to resolwe-bio repository
+        :rtype: dict
+        """
+        all_process_files = []
+        process_file_extensions = ['*.yaml', '*.yml']
+        for root, _, filenames in os.walk(process_dir):
+            for extension in process_file_extensions:
+                for filename in fnmatch.filter(filenames, extension):
+                    all_process_files.append(os.path.join(root, filename))
+
+        def read_yaml_file(fname):
+            with open(fname) as file_:
+                return yaml.load(file_)
+
+        all_processes = {}
+        for yaml_file in all_process_files:
+            processes_in_file = read_yaml_file(yaml_file)
+            for process in processes_in_file:
+
+                # This section finds the line in file where the
+                # defintion of the process starts. (there are
+                # multiple process definition in some files).
+                startline = self.get_process_definition_start(yaml_file, process['slug'])
+
+                # Put together URL to starting line of process definition.
+                cutted_yaml_path = yaml_file.split('/resolwe_bio/processes/')
+                source_base_url = source_base_url + cutted_yaml_path[1] + '#L' + str(startline)
+                chosen_process = [proc for proc in processes_in_file if proc['slug'] == process['slug']]
+                assert len(chosen_process) == 1
```

Comment:
What happens if this is not `True`? Add message with the description of the error.